std/traits.d

```diff
@@ -5563,15 +5563,38 @@ enum bool isSomeChar(T) = is(CharTypeOf!T) && !isAggregateType!T;
     static assert(!isSomeChar!(S!char));
 }
 
-/**
-Detect whether $(D T) is one of the built-in string types.
-
-The built-in string types are $(D Char[]), where $(D Char) is any of $(D char),
-$(D wchar) or $(D dchar), with or without qualifiers.
-
-Static arrays of characters (like $(D char[80])) are not considered
-built-in string types.
- */
+/++
+    Detect whether `T` is one of the built-in string types.
+
+    A built-in string type is any array type whose element type is `char`,
+    `wchar`, or `dchar` - regardless of qualifiers (that includes `string`,
+    `wstring`, and `dstring`).
+
+    All arrays whose element types are `char`, `wchar`, or `dchar` -
+    regardless of qualifiers (that includes `string` and `wstring`). It
+    specifically does $(I not) include arrays of `dchar`.
+
+    Static arrays of characters (like `char[80]`) are not considered built-in
+    string types.
+
+    $(RED Warning:
+        This trait is true for enums with a base type that is a built-in string
+        type. In some cases, this is useful, but in many cases, for a function
+        to work correctly with both an enum and the enum's base type, the enum
+        value needs to be converted to the base type (which won't happen
+        automatically if the function is templated on the parameter type rather
+        than expicitly accepting the enum's base type); otherwise, it risks not
+        compiling or ending up with a value that is not one of the members of
+        that enum. So, code that uses isSomeString should be careful to ensure
+        that it either works with enums or that it disallows them - e.g.
+        `isSomeString!T && !is(T == enum)`. In general, if a function is going
+        to accept any string type rather than a generic range of characters,
+        it's simpler and safer to just templatize on character type rather than
+        to use isSomeString. However, in other cases, isSomeString can be quite
+        useful when used in combination with other traits. Just be aware that
+        it will be true for enums whose base type is a string type and code
+        accordingly.)
```

Comment:
These last 2 sentences seem to be a repetition of what was already said -- perhaps write a shorter summary? Or take them out altogether.